biz.aQute.bndlib/src/aQute/bnd/build/Project.java

```diff
@@ -2112,6 +2112,15 @@ public void test(File reportDir, List<String> tests) throws Exception {
 			IO.delete(reportDir);
 			tester.setReportDir(reportDir);
 		}
+		if (tests == null) {
+			// check system properties: gradle
+			String singleTest = System.getProperty("integTest.single");
```

Comment:
> are you suggesting we just use yet another system property specific to bnd's test infra?

No. I am suggesting it is done properly in the proper place. And here is not the proper place and it is not done in the proper way.